src/ssl.c

```diff
@@ -7970,7 +8090,275 @@ int ProcessBuffer(WOLFSSL_CTX* ctx, const unsigned char* buff,
+                /* Dilithium is fixed key size */
+                keySz = DILITHIUM_LEVEL2_KEY_SIZE;
+                if (ssl && !ssl->options.verifyNone) {
+                    if (ssl->options.minDilithiumKeySz < 0 ||
+                          keySz < (int)ssl->options.minDilithiumKeySz) {
+                        ret = DILITHIUM_KEY_SIZE_E;
+                        WOLFSSL_MSG("Certificate Dilithium key size error");
+                    }
+                }
+                else if (ctx && !ctx->verifyNone) {
+                    if (ctx->minDilithiumKeySz < 0 ||
+                                  keySz < (int)ctx->minDilithiumKeySz) {
+                        ret = DILITHIUM_KEY_SIZE_E;
+                        WOLFSSL_MSG("Certificate Dilithium key size error");
+                    }
+                }
+                break;
+            case DILITHIUM_LEVEL3k:
+                keyType = dilithium_level3_sa_algo;
+                /* Dilithium is fixed key size */
+                keySz = DILITHIUM_LEVEL3_KEY_SIZE;
+                if (ssl && !ssl->options.verifyNone) {
+                    if (ssl->options.minDilithiumKeySz < 0 ||
+                          keySz < (int)ssl->options.minDilithiumKeySz) {
+                        ret = DILITHIUM_KEY_SIZE_E;
+                        WOLFSSL_MSG("Certificate Dilithium key size error");
+                    }
+                }
+                else if (ctx && !ctx->verifyNone) {
+                    if (ctx->minDilithiumKeySz < 0 ||
+                                  keySz < (int)ctx->minDilithiumKeySz) {
+                        ret = DILITHIUM_KEY_SIZE_E;
+                        WOLFSSL_MSG("Certificate Dilithium key size error");
+                    }
+                }
+                break;
+            case DILITHIUM_LEVEL5k:
+                keyType = dilithium_level5_sa_algo;
+                /* Dilithium is fixed key size */
+                keySz = DILITHIUM_LEVEL5_KEY_SIZE;
+                if (ssl && !ssl->options.verifyNone) {
+                    if (ssl->options.minDilithiumKeySz < 0 ||
+                          keySz < (int)ssl->options.minDilithiumKeySz) {
+                        ret = DILITHIUM_KEY_SIZE_E;
+                        WOLFSSL_MSG("Certificate Dilithium key size error");
+                    }
+                }
+                else if (ctx && !ctx->verifyNone) {
+                    if (ctx->minDilithiumKeySz < 0 ||
+                                  keySz < (int)ctx->minDilithiumKeySz) {
+                        ret = DILITHIUM_KEY_SIZE_E;
+                        WOLFSSL_MSG("Certificate Dilithium key size error");
+                    }
+                }
+                break;
+        #endif /* HAVE_DILITHIUM */
+        #endif /* HAVE_PQC */
+
+            default:
+                WOLFSSL_MSG("No alt key size check done on certificate");
+                break; /* do no check if not a case for the key */
```

Comment:
Does this handle the case where an sapkiOID was set but the keysize switch case may be compiled out? Want to make sure that there is not the possibility of skipping over a check on the key.

Reply:
In that case keySz and keyType would both remain 0.  That's good and desired behavior. IMHO. 

Ah....however, ret would not get set properly in that case.  I'll look into a fix.

Reply:
My solution is to set  ret to NOT_COMPILED_IN in default case.  See the next commit.